compute/cker/include/cker/operation/Tanh.h

```diff
@@ -36,6 +37,40 @@ inline void Tanh(const Shape &input_shape, const float *input_data, const Shape
   output_map.array() = input_map.array().tanh();
 }
 
+inline void Quant8_tanh(int32_t input_zero_point, int32_t input_range_radius,
+                        int32_t input_multiplier, int32_t input_shift,
+                        int32_t input_size, const int8_t *input_data,
+                        int8_t *output_data)
+{
+  static constexpr int32_t kInputIntegerBits = 4;
```

Comment:
This implementation appears to come from tflite `reference_interger_ops`, which is for `int16_t`. The implementation from `optimized_ops` needs to be ported. Could you please check once?

Reply:
I think tensorflow has two implementations of tanh function to support int8 type.

the first one is placed in tensorflow/lite/kernels/activation.cc -> Tanhval function

the second is placed in tensorflow/lite/micro/kernels/tanh.cc -> Tanhval function

I don't know what is diffrence of first and second.

if you know about it, could you explain diffrence between them ?

Reply:
I guess the difference between first and second is the existence of a lookup table.